src/compiler/scala/tools/nsc/transform/OverridingPairs.scala

```diff
@@ -53,9 +53,6 @@ abstract class OverridingPairs extends SymbolPairs {
       && !exclude(low)                 // this admits private, as one can't have a private member that matches a less-private member.
       && (lowMemberType matches (self memberType high))
     ) // TODO we don't call exclude(high), should we?
-
-    override def skipOwnerPair(lowClass: Symbol, highClass: Symbol): Boolean =
-      lowClass.isJavaDefined && highClass.isJavaDefined // javac is already checking this better than we could
```

Comment:
In Dotty I ended up with a similar check but restricted to the case where both symbols are actual classes and not interfaces: https://github.com/lampepfl/dotty/blob/69c800b22c0ae6c5eb6fa6e7c2142c18a8278ff1/compiler/src/dotty/tools/dotc/typer/RefChecks.scala#L506 since one can't extend two Java classes without one extending the other, but I'm not sure if it's completely correct either.